packages/code-du-travail-modeles/src/modeles/conventions/hospitalisation_privees.yaml

```diff
@@ -0,0 +1,43 @@
+contrat salarié . convention collective . hospitalisation privées:
+  description: Convention collective nationale de l'hospitalisation privée du 18 avril 2002.
+  applicable si: convention collective = 'IDCC2264'
+  valeur: oui
+  références:
+    Article 50.3: https://www.legifrance.gouv.fr/conv_coll/id/KALIARTI000005802016/?idConteneur=KALICONT000005635813
+  cdtn:
+    type: "overrideSeniority"
```

Comment:
Après lecture, je pense que l'ajout de la complexité dans le YAML n'est peut être pas la solution.

On a une implémentation UI spécifique. Soit on décide qu'elle doit se refléter dans publicodes, soit on en fait abstraction.

Aujourd'hui, pour l'ancienneté, on en a fait abstraction. A savoir que l'on a une règle ancienneté dans le code qui attend une ancienneté en mois et une UI qui abstrait la notion de plus de 24 mois ou moins.

Là, on inclus une règle très technique dans publicode pour gérer le cas où le salarié à plus de 5 ans. Pourquoi dans le cas présent, publicode porte l'information ? 

Pour moi, il faut rester sur la même logique dans les 2 cas pour éviter de l'incompréhension futur. Dans le cas présent, je vois 2 solutions.

### C'est de l'UI

Dans ce cas, publicode ne le gère pas. On va ajouter la règle dans le code react qui va intégrer cette question si on est sur la CC 2264. 

Avantages:
 * La règle publicode reste simple
 * La règle est en clair dans react et on peut y apposer des TU facilement
Inconvénients:
 * On ne gère que ce cas, si demain une autre condition sur l'ancienneté arrive, il n'y a pas de mécanisme spécifique.

### Publicode to rule them all

Dans ce cas, actuellement, la gestion de l'ancienneté n'est pas seine. Il faudrait avoir une règle publicode qui demande si le salarié à plus de 2 ans puis si il a moins de 2 ans, un règle qui demande combien il a exactement.
Dans ce cas là, on peut facilement ajouter une règle pour cette CC qui va demander si + ou - de 5 ans en plus.

Exemple:

```yaml
contrat salarié . ancienneté plus de 2 ans:
  titre: Ancienneté du salarié plus de 2 ans
  question: Le salarié a-t-il plus de 2 ans d'ancienneté dans l'entreprise (2 ans + 1 jour) ?
  cdtn:
    type: oui-non

contrat salarié . ancienneté moins de 2 ans:
  applicable si: ancienneté plus de 2 ans = non
  titre: Ancienneté du salarié en mois
  question: Quelle est l'ancienneté du salarié dans l’entreprise en mois ?
  unité: mois
  cdtn:
    type: entier

contrat salarié . ancienneté:
  titre: Ancienneté du salarié
  unité: mois
  variations:
     - si: ancienneté plus de 2 ans
        alors: 25 mois
     - sinon: ancienneté moins de 2 ans
```

Et pour cette CC, on va poser la question supplémentaire:
```
contrat salarié . convention collective . hospitalisation privées . ancienneté plus de 5 ans:
  applicable si: ancienneté plus de 2 ans
  titre: Ancienneté du salarié plus de 5 ans
  question: Le salarié a-t-il plus de 5 ans d'ancienneté dans l'entreprise (5 ans + 1 jour) ?
  cdtn:
    type: oui-non

contrat salarié . convention collective . hospitalisation privées . ancienneté :
   applicable si: ancienneté plus de 5 ans
  remplace: ancienneté
  par: 60 mois
```

Ici, publicode dirige l'ensemble des questions liées à l'ancienneté et on peut gérer les cas particulier sans devoir passer par du code propre à CDTN.

Je n'ai pas d'avis tranché sur la question sur le choix à faire. La solution 1 me semble logique car publicode devrait s'abstraire un maximum du comportement de l'UI et être au plus simple mais en même temps ça évite d'en avoir partout. La solution 2 me plait car on retrouve ce que l'on fait dans l'UI dans notre publicode.
Par contre, un mélange des 2 sur le même sujet, je trouve cela déroutant.

Reply:
Mettre la question en statique dans le code frontend.
Trigger avec le numéro de la CC.